### PR TITLE
Define schema with Scala classes

### PR DESCRIPTION
**Pull request by pstutz**
Implementation of #110

### CODE REVIEW

**Comment by pstutz** [comment]
Implementation of #110

File: src/main/scala/org/opencypher/caps/api/record/GraphScan.scala [hunk 1]
```diff
@@ -48,6 +51,15 @@ sealed trait GraphScan extends Serializable {
 }
 
 object GraphScan extends GraphScanCompanion[EmbeddedEntity] {
+  implicit def nodesToScan[E <: Node : TypeTag](
+    nodes: Seq[E])(implicit caps: CAPSSession): NodeScan = {
```
Comment (Mats-SX) [inline src/main/scala/org/opencypher/caps/api/record/GraphScan.scala]: This linebreak seems a little off

File: src/main/scala/org/opencypher/caps/api/record/GraphScan.scala [hunk 2]
```diff
@@ -48,6 +51,15 @@ sealed trait GraphScan extends Serializable {
...
+    NodeScan(nodes)
+  }
+
+  implicit def relationshipsToScan[E <: Relationship : TypeTag](
+    relationship: Seq[E])(implicit caps: CAPSSession): RelationshipScan = {
```
Comment (Mats-SX) [inline src/main/scala/org/opencypher/caps/api/record/GraphScan.scala]: Also this one

File: src/main/scala/org/opencypher/caps/api/record/GraphScan.scala (hunk 1 above)
Comment (pstutz) [reply src/main/scala/org/opencypher/caps/api/record/GraphScan.scala]: I agree. Oddly enough IntelliJ defaults with our changes seems to like it that way and reformats it that way when I change it.

Maybe we can automate style enforcement the same way we automate the license header formatting.

**Pull request merged**